workflow_zstack/paths.py

```diff
@@ -0,0 +1,75 @@
+from collections.abc import Sequence
+from typing import List
+import pathlib
+import datajoint as dj
+from element_interface.utils import find_full_path
+from element_session import session_with_id as session
+
+
+def get_session_directory(session_key: dict) -> str:
+    """Return relative path from SessionDirectory table given key
+
+    Args:
+        session_key (dict): Key uniquely identifying a session
+
+    Returns:
+        path (str): Relative path of session directory
+    """
+    from .pipeline import session
+
+    # NOTE: fetch (vs. fetch1) permits dir to not exist, may be the case when saving
+    # slices directly from from BossDB into inferred dir based on BossDB structure
+    session_dir = (session.SessionDirectory & session_key).fetch("session_dir")
+
+    if len(session_dir) > 1:
+        raise ValueError(
+            f"Found >1 directory for this key:\n\t{session_key}\n\t{session_dir}"
+        )
+    elif len(session_dir) == 1:
+        return session_dir[0]
+    else:
+        return None
```

Comment:
The primary key for `session.SessionDirectory` is `-> Session`, so it is not possible for multiple directories to be returned.

Reply:
This function isn't necessary in the workflow and has been removed.